test/strtoultest.c

```diff
@@ -36,7 +36,7 @@ static struct strtoul_test_entry strtoul_tests[] = {
     {
         "0x12345", 10, 0, 1, 1
     },
-#if defined(__WORDSIZE) && __WORDSIZE == 64
+#if ULONG_MAX == 18446744073709551615UL
```

Comment:
Ah - the reason for suggesting this the other way round (using the explicit 32-bit ULONG_MAX value and having the else be the 64-bit case) is in case the pre-processor on some 32-bit system doesn't handle 64-bit integers very well. I am confident that on both 32- and 64-bit systems the pre-processor will be able to handle calculations up to 4294967295, but not the other way round

Reply:
Ok, changed it to test for 32 bit values, and reworded the original commit